rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -527,10 +531,13 @@ def _decode_transaction(
                 events.append(decoding_output.event)
                 continue
 
-            token = GlobalDBHandler.get_evm_token(
-                address=tx_log.address,
-                chain_id=self.evm_inquirer.chain_id,
-            )
+            if (token := tokens_cache.get(tx_log.address)) is None:
+                token = GlobalDBHandler.get_evm_token(
+                    address=tx_log.address,
+                    chain_id=self.evm_inquirer.chain_id,
+                )
+                tokens_cache[tx_log.address] = token
```

Comment:
instead of a temporary cache like this here, would it not be better to do `get_or_create_evm_token`? Don't we have a cache somewhere there already in the resolver? This is why we want to be using that function over anything else.

I guess your cache for this particular case is a lot more straighforward but don't we already do that in get_or_create() with an in memory LRU cache? I remember we did. And if we do not then we absolutely should as this would solve similar problems in smaller scale over the rest of the codebase.